api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto

```diff
@@ -453,6 +453,26 @@ message HttpConnectionManager {
   // with `prefix` match set to `/dir`. Defaults to `false`. Note that slash merging is not part of
   // `HTTP spec <https://tools.ietf.org/html/rfc3986>` and is provided for convenience.
   bool merge_slashes = 33;
+
+  // // Determines if adjacent slashes in the path are merged into one before any processing of
+  // // requests by HTTP filters or routing. This affects the upstream *:path* header as well. Without
+  // // setting this option, incoming requests with path `//dir///file` will not match against route
+  // // with `prefix` match set to `/dir`. Defaults to `false`. Note that slash merging is not part of
+  // // `HTTP spec <https://tools.ietf.org/html/rfc3986>` and is provided for convenience.
+  // bool merge_slashes = 33;
+
+  // Configuration of local reply send by envoy for http responses.
```

Comment:
I think I'm missing something. How does this configuration work exactly?

/wait-any

Reply:
I see this makes sense and is pretty click. But don't we still then potentially need matching and rewriting as a parallel set of options? cc @ahedberg @lizan for comment

/wait-any

Reply:
I've added mapper config to allows `response_code` mapping based on `response_flags`. I've reused `AccessLogFilter` to achieve that. Here is sample config which I tested:
```
              local_reply_config:
                mapper:
                  matcher:
                    response_flag_filter:
                      flags:
                        - LH
                        - UH
                  rewriter:
                    status_code: 504
                json_format:
                      foo: "test"
                      timestamp: "%START_TIME(%FT%T.%3fZ)%"
                      level: TRACE
                      logger: envoy
                      request_protocol: "%PROTOCOL%"
                      request_method: "%REQ(:METHOD)%"
                      request_authority: "%REQ(:authority)%"
                      request_path: "%REQ(:PATH)%"
                      response_body: "%RESP_BODY%"
```

Reply:
the config looks reasonable, any chance the matcher part can be unified with access logger filter?

for the rewriter part we may need support grpc-status / grpc-message / grpc-status-details-bin, but you may leave that as a TODO for now.

Reply:
+1 to merging with the access log filter matching. 

In general though this looks like a great path forward to me.

Reply:
+1 as well on merging with existing matchers.

Reply:
Changed to AccessLogFilter. If it looks ok for you I will try to implement this and if I have problems I will ask for some guidance.

Reply:
@lukidzi can you potentially just cleanup this PR and push only the final API and docs? I would like us to agree on that before implementation. Thank you!

Reply:
Sorry for the delay. I think the current design should let us achieve what I was requesting in #8214. cc @alyssawilk

Reply:
Added docs and removed rest of changes. If I should be more specific and place them somewhere else I will fix it in next commit. Thanks !

Reply:
Yeah, I think this hits 2 things we want - custom error codes and custom error messages.
My one concern is we might have the knobs we need, but if the flags aren't enough maybe we can add in the response details into matcher options.